source/s3_auto_ranged_get.c

```diff
@@ -182,6 +182,39 @@ static void s_s3_auto_ranged_get_next_request(
         }
     }
 
+    /* One sepcial error case is an empty file download */
```

Comment:
not directly related, but this function is turning into a nightmare and should probably be split/refactored if possible.

I feel like a lot of this could be moved into a helper function.